include/alpaka/dev/DevOacc.hpp

```diff
@@ -0,0 +1,341 @@
+                    vspQueues.reserve(m_queues.size());
+
+                    for(auto it = m_queues.begin(); it != m_queues.end();)
+                    {
+                        auto spQueue(it->lock());
+                        if(spQueue)
+                        {
+                            vspQueues.emplace_back(std::move(spQueue));
+                            ++it;
+                        }
+                        else
+                        {
+                            it = m_queues.erase(it);
+                        }
+                    }
+                    return vspQueues;
+                }
+
+                //-----------------------------------------------------------------------------
+                //! Registers the given queue on this device.
+                //! NOTE: Every queue has to be registered for correct functionality of device wait operations!
+                ALPAKA_FN_HOST auto registerQueue(std::shared_ptr<IGenericThreadsQueue<DevOacc>> spQueue)
+                -> void
+                {
+                    std::lock_guard<std::mutex> lk(m_Mutex);
+
+                    // Register this queue on the device.
+                    m_queues.push_back(std::move(spQueue));
+                }
+
+                int iDevice() const {return m_iDevice;}
+                acc_device_t deviceType() const {return m_deviceType;}
+
+            private:
+                std::mutex mutable m_Mutex;
+                std::vector<std::weak_ptr<IGenericThreadsQueue<DevOacc>>> mutable m_queues;
+                acc_device_t m_deviceType;
+                int m_iDevice;
+            };
+        }
+    }
+    //#############################################################################
+    //! The OpenACC device handle.
+    class DevOacc :
+        public concepts::Implements<ConceptCurrentThreadWaitFor, DevOacc>,
+        public concepts::Implements<ConceptDev, DevOacc>
+    {
+        friend struct traits::GetDevByIdx<PltfOacc>;
+
+    protected:
+        //-----------------------------------------------------------------------------
+        DevOacc(int iDevice) :
+            m_spDevOaccImpl(std::make_shared<oacc::detail::DevOaccImpl>(iDevice))
+        {}
+    public:
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc const &) = default;
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc &&) = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc const &) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc &&) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator==(DevOacc const & rhs) const
```

Comment:
Just because it noticed it here. What is the rational for explicitely marking a function as `ALPAKA_FN_HOST`? We could just omit that and it would have the same meaning and be shorter?

Reply:
It is true, that `ALPAKA_FN_HOST` is just empty for OpenACC. This is intended in part documentation ("do not try using this in device code, which we cannot check") and mostly for consistency with other backend, where `ALPAKA_FN_HOST` actually does something  (i.e. CUDA).

Reply:
But even for CUDA, would it make a difference in semantic? AFAIK you cannot call an unannotated function from a device function as well.